src/mon/Monitor.cc

```diff
@@ -2631,7 +2631,19 @@ void Monitor::handle_command(MonOpRequestRef op)
     return;
   }
 
-  cmd_getval(g_ceph_context, cmdmap, "prefix", prefix);
+  bool valid_cmd = cmd_getval(g_ceph_context, cmdmap, "prefix", prefix);
+
+  // check return value and prefix.
+  if (prefix.empty()) {
+    reply_command(op, -EINVAL, "empty command found", 0);
+    return;
+  }
+
+  if (!valid_cmd) {
```

Comment:
I suggest moving this check just below the cmd_getval() call. No need to check if prefix is empty if the call failed and it will better outline the what comes before what.